probe/websocket/ws_test.go

```diff
@@ -50,6 +51,21 @@ func TestWSPing(t *testing.T) {
 		}
 	}()
 
+	// wait for http server to start
+	conn, err := net.Conn(nil), error(nil)
+	for i := 0; i < 30; i++ {
+		conn, err = net.Dial("tcp", "127.0.0.1:18080")
```

Comment:
Consider handling the error from the `net.Dial` function inside the loop. This way, if the server fails to start, the error can be logged and the test can be stopped immediately. [important]